mantidproject/mantid
# Add automatic conversions for unicode to std::string in Python 2

**Description of work.**

Registers an additional converter for Python 2 that allows a `std::string` to be extracted from a `unicode` object automatically by the `boost::python` framework. The `unicode` object is encoded to a bytes object using "utf-8" and then extracted to `std::string`. Any encoding errors will be raised to Python as normal.

This also removes the need for the manual proxy methods that were doing this behaviour at each separate extraction point.

**Report to:** [nobody]. 

**To test:**

Verify that the examples in the referenced issues work in the workbench script editor.

Check the unit tests cover the appropriate methods.

Fixes #23959 
Fixes #23964 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
